wikiscraper.py

```diff
@@ -1,175 +1,264 @@
-def stripSyntaxHighlighter(text):
-    text = text.replace("<syntaxhighlight lang=\"javascript\">", "")
-    text = text.replace("</syntaxhighlight>", "")
-
-    return text
-
-def hashPageText(data):
-    data["textHash"] = hashlib.sha256(data["text"].encode("utf-8")).hexdigest()
-    del data["text"]
-
-def combineNamesAndDescriptions(d, varType):
-    nameField = varType + "VariableNames"
-    descField = varType + "VariableDescriptions"
-    finalField = varType + "Variables"
-    if nameField in d and descField in d:
-        if len(d[nameField]) == 0 or len(d[descField]) == 0:
-            print "Empty array found!"
-        elif len(d[nameField]) != len(d[descField]):
-            print "Unmatched pairs!"
+    def getResponseFileName(self, index):
+        return self.configFileName.replace(".yml", "-raw-%03d.json" % index)
+
+    def loadResponseFile(self, index):
+        responseFileName = self.getResponseFileName(index)
+        if os.path.isfile(responseFileName):
+            with open(responseFileName, "r") as responseFile:
+                return json.load(responseFile)
         else:
-            pairs = {}
-            pattern = re.compile("\[\[([A-Za-z]*)\|([A-Za-z]*)\]\]")
-            for i in xrange(0, len(d[nameField])):
-                key = d[nameField][i]
-                val = d[descField][i]
-                m = pattern.match(key)
-                if m:
-                    pairs[m.group(2)] = val + " (See %s Wikia page for more details.)" % m.group(1)
-                else:
-                    pairs[key] = val
-            d[finalField] = pairs
-            del d[nameField]
-            del d[descField]
-    elif nameField in d:
-        print "Only names found!"
-    elif descField in d:
-        print "Only descriptions found!"
-
-def cleanExampleRequest(data):
-    if "httpMethod" in data and "exampleRequest" in data and data["httpMethod"] == "post":
-        try:
-            req = json.loads(data["exampleRequest"].split("\n", 1)[1])
-            data["exampleRequest"] = json.dumps(req)
-        except ValueError as ve:
-            print "Error decoding JSON for exampleRequest in " + data["name"]
-
-# GetPublicXurVendor has blocks for "When Xur is/isn't available."
-def cleanExampleResponse(data):
-    if data["name"] == "GetPublicXurVendor":
-        res = data["exampleResponse"].replace("When Xur isn't available.", "")
-        responses = res.split("When Xur is available.")
-        try:
-            data["exampleResponses"] = [
-                json.dumps(json.loads(responses[1])),
-                json.dumps(json.loads(responses[0]))
+            raise IOError("Response file %s does not exist" % responseFileName)
+
+    # TODO: Rip this out and put it in a custom domain module
+    def stripSyntaxHighlighter(self, text):
+        text = text.replace("<syntaxhighlight lang=\"javascript\">", "")
+        text = text.replace("</syntaxhighlight>", "")
+
+        return text
+
+    def hashPageText(self, data):
```

Comment:
I'm uncomfortable with this mutation. I'd suggest mutation a copy of `data` and returning that instead?